File: src/main/java/com/github/zametki/db/dbi/ZametkaDbi.java
package com.github.zametki.db.dbi;

import com.github.zametki.model.GroupId;
import com.github.zametki.model.UserId;
import com.github.zametki.model.Zametka;
import com.github.zametki.model.ZametkaId;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface ZametkaDbi {

    void create(@NotNull Zametka zametka);

    /**
     * Список заметок сортированный по дате.
     */
    @NotNull
    List<ZametkaId> getByUser(@Nullable UserId userId);

    @Nullable
    Zametka getById(@Nullable ZametkaId id);

    void delete(@NotNull ZametkaId zametkaId);

    void update(@NotNull Zametka z);

    int countByCategory(UserId userId, @NotNull GroupId groupId);
}

File: src/main/java/com/github/zametki/db/dbi/ZametkaDbi.kt
package com.github.zametki.db.dbi

import com.github.zametki.model.GroupId
import com.github.zametki.model.UserId
import com.github.zametki.model.Zametka
import com.github.zametki.model.ZametkaId

interface ZametkaDbi {

    fun create(zametka: Zametka)

    /**
     * Список заметок сортированный по дате.
     */
    fun getByUser(userId: UserId?): List<ZametkaId>

    fun getById(id: ZametkaId?): Zametka?

    fun delete(zametkaId: ZametkaId)

    fun update(z: Zametka)

    fun countByGroup(userId: UserId, groupId: GroupId): Int
}
